Add documentation for DenonAVR telnet feature

dcmeglio opened the pull request:
## Proposed change
<!-- 
    Describe the big picture of your changes here to communicate to the
    maintainers why we should accept this pull request. If it fixes a bug
    or resolves a feature request, be sure to link to that issue in the 
    additional information section.
-->

Adds a couple more supported AVRs to the list (based on my personal testing), fixed some mistakes related to zone 1/zone 2 really should be zone 2/zone 3, added documentation for new telnet feature.

## Type of change
<!--
    What types of changes does your PR introduce to our documentation/website?
    Put an `x` in the boxes that apply. You can also fill these out after
    creating the PR.
-->

- [ ] Spelling, grammar or other readability improvements (`current` branch).
- [ ] Adjusted missing or incorrect information in the current documentation (`current` branch).
- [ ] Added documentation for a new integration I'm adding to Home Assistant (`next` branch).
  - [ ] I've opened up a PR to add logo's and icons in [Brands repository](https://github.com/home-assistant/brands).
- [X] Added documentation for a new feature I'm adding to Home Assistant (`next` branch).
- [ ] Removed stale or deprecated documentation.

## Additional information
<!--
    Details are important, and help maintainers processing your PR.
    Please be sure to fill out additional details, if applicable.
-->

- Link to parent pull request in the codebase: https://github.com/home-assistant/core/pull/85980
- Link to parent pull request in the Brands repository: 
- This PR fixes or closes issue: fixes #

## Checklist
<!--
    Put an `x` in the boxes that apply. You can also fill these out after
    creating the PR. If you're unsure about any of them, don't hesitate to ask.
    We're here to help! This is simply a reminder of what we are going to look
    for before merging your code.
-->

- [X] This PR uses the correct branch, based on one of the following:
  - I made a change to the existing documentation and used the `current` branch.
  - I made a change that is related to an upcoming version of Home Assistant and used the `next` branch.
- [X] The documentation follows the Home Assistant documentation [standards].

[standards]: https://developers.home-assistant.io/docs/documenting/standards

dcmeglio committed:
Add documentation for telnet feature.

frenck reviewed (dismissed):
✅ Approved. Can be merged as soon as the parent PR gets merged.

dcmeglio committed:
Change default for use_telnet to True and add note thatit only applies to new installs

dcmeglio committed:
Merge branch 'home-assistant:next' into next

dcmeglio commented:
@frenck per some PR reviews on the parent, I've changed the default value of `use_telnet` to true. To prevent a breaking change though it will remain false for existing installs but will be true if you setup a new install. I added text to hopefully make this clear but let me know if there is a better way.

dcmeglio committed:
Update denonavr documentation based on core PR feedback changes.
